src/plonk/circuit.rs

```diff
@@ -780,9 +784,36 @@ impl<F: Field> ConstraintSystem<F> {
         // Account for each gate to ensure our quotient polynomial is the
         // correct degree and that our extended domain is the right size.
         for (_, poly) in self.gates.iter() {
-            degree = std::cmp::max(degree, poly.degree());
+            // All gates are multiplied by (1 - l_cover(X))
+            degree = std::cmp::max(degree, poly.degree() + 1);
         }
 
         degree
     }
+
+    /// Compute the number of blinding factors necessary to perfectly blind
+    /// each of the prover's witness polynomials.
+    pub fn blinding_factors(&self) -> usize {
+        // All of the prover's advice columns are evaluated at most
+        let factors = *self.num_advice_queries.iter().max().unwrap_or(&1);
+        // distinct points during gate checks. In the permutation and lookup
+        // argument the witness polynomials are evaluated at most 2 times:
+
+        let factors = std::cmp::max(2, factors);
+
+        // Each polynomial is evaluated at most an additional time during
+        // multiopen
+        let factors = factors + 1;
+
+        // h(x) is derived by the other evaluations so it does not reveal
+        // anything; in fact it does not even appear in the proof.
+
+        // h(x_3) is also not revealed; the verifier only learns a single
+        // evaluation of a polynomial in x_1 which has h(x_3) and another random
+        // polynomial evaluated at x_3 as coefficients
+
+        // Add an additional blinding factor so that each polynomial becomes
+        // computationally indistinguishable from random
+        factors + 1
+    }
```

Comment:
```suggestion
    }

    /// Compute the number of inactive rows needed
    pub fn inactive(&self) -> usize {
        ConstraintSystem::blinding_factors(&self) + 1
    }
```

Reply:
this saves us from having to do `blinding_factors + 1` or `- blinding_factors - 1` in various places.